sound/soc/sof/ipc4-topology.c

```diff
@@ -495,8 +503,27 @@ static int sof_ipc4_widget_setup_comp_dai(struct snd_sof_widget *swidget)
 
 			blob->alh_cfg.count++;
 		}
+		if (blob->alh_cfg.count > 1) {
+			for (i = 0; i < ALH_MULTI_GTW_COUNT; i++) {
+				if (alh_groups[i].generated &&
+				    !strncmp(swidget->widget->sname, alh_groups[i].sname,
+				    ALH_SNAME_SIZE)) {
+					blob->group_id = ALH_MULTI_GTW_BASE + i;
+					break;
+				}
+				if (!alh_groups[i].generated) {
+					blob->group_id = ALH_MULTI_GTW_BASE + i;
+					alh_groups[i].generated = true;
+					strncpy(alh_groups[i].sname, swidget->widget->sname,
+						ALH_SNAME_SIZE);
+					break;
+				}
+			}
+		}
```

Comment:
This looks like manually handling an IDA...

what I don't get is the earlier comment that  'I didn't figure out where should we save the ida'. We have multiple cases where the IDA is a static variable in the file. see for example drivers/soundwire/bus.c

````c
static DEFINE_IDA(sdw_ida);

static int sdw_get_id(struct sdw_bus *bus)
{
	int rc = ida_alloc(&sdw_ida, GFP_KERNEL);

	if (rc < 0)
		return rc;

	bus->id = rc;
	return 0;
}
````